.github/workflows/fe-build-and-test.yaml

```diff
@@ -0,0 +1,34 @@
+name: FE Test and Build
+
+on:
+  push:
+    branches-ignore:
+      - master
+    paths:
+      - thirdeye-ui/**
+  workflow_dispatch:
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Pull Repository
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            thirdeye-ui
+
+      - name: Install node and npm
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14
+          cache: "npm"
+          cache-dependency-path: './thirdeye-ui/package-lock.json'
+
+      - name: Install dependencies, run test, and build
```

Comment:
Wouldn't it be better to do this in separate steps? Can you please check if, when you create separate steps, the installed `node_modules` gets cached in the GitHub action's artifacts?